build_tools/teatime.py

```diff
@@ -103,6 +110,60 @@ def finish(self):
                 b"[" + self.label + b" completed in " + run_pretty.encode() + b"]\n"
             )
 
+        # Return if upload is not configured
+        if not (self.upload_to_s3 and self.s3_bucket and self.s3_subdir):
+            return
+
+        repo_root = Path(__file__).resolve().parent.parent
+        build_dir = repo_root / self.base_build_dir
+        amdgpu_family = os.getenv("AMDGPU_FAMILIES")
+
+        # Step 1: Run create_log_index.py (if AMDGPU_FAMILIES is defined)
+        if amdgpu_family:
+            try:
+                index_script = repo_root / "build_tools" / "create_log_index.py"
+                print(f"[TEATIME] Indexing logs for AMDGPU_FAMILIES={amdgpu_family}")
+                subprocess.run(
+                    [
+                        sys.executable,
+                        str(index_script),
+                        "--build-dir",
+                        str(build_dir),
+                        "--amdgpu-family",
+                        amdgpu_family,
+                    ],
+                    check=True,
+                )
+            except subprocess.CalledProcessError as e:
+                print(f"[WARN] create_log_index.py failed: {e}", file=sys.stderr)
+            except Exception as e:
+                print(
+                    f"[WARN] Unexpected error during log indexing: {e}", file=sys.stderr
+                )
+        else:
+            print("[WARN] AMDGPU_FAMILIES not set; skipping log indexing")
+
+        # Step 2: Upload logs to S3
+        try:
+            upload_script = repo_root / "build_tools" / "upload_logs_to_s3.py"
+            s3_path = f"s3://{self.s3_bucket}/{self.s3_subdir}"
```

Comment:
I would rather pass the explicit arguments instead of fusing and splitting them later again. See other comment.